sql/src/main/java/io/crate/analyze/relations/RelationNormalizer.java

```diff
@@ -66,13 +63,30 @@ protected AnalyzedRelation visitAnalyzedRelation(AnalyzedRelation relation, Coor
         }
 
         @Override
-        public AnalyzedRelation visitQueriedSelectRelation(QueriedSelectRelation relation, CoordinatorTxnCtx context) {
+        public AnalyzedRelation visitQueriedSelectRelation(QueriedSelectRelation<?> relation, CoordinatorTxnCtx context) {
             AnalyzedRelation subRelation = relation.subRelation();
-            AnalyzedRelation normalizedSubRelation = process(relation.subRelation(), context);
-            if (subRelation == normalizedSubRelation) {
-                return relation;
+            AnalyzedRelation normalizedSubRelation = process(subRelation, context);
+
+
+            QueriedSelectRelation<?> newRelation;
+            if (normalizedSubRelation instanceof FieldResolver) {
+                EvaluatingNormalizer evalNormalizer = new EvaluatingNormalizer(
+                    functions, RowGranularity.CLUSTER, null, ((FieldResolver) normalizedSubRelation));
+
+                if (subRelation == normalizedSubRelation) {
```

Comment:
I'm inclined to drop these optimizations in favour of having just one code path. I doubt it is worth it. At least the inner most relation is usually normalized because of the Fields -> Reference conversion, so all ancestors will in turn also be normalized. So I think the optimized "no-op" path hardly ever kicks in.

Thoughts?

Reply:
Yeah, I agree it's likely not winning us any time, just complicates the code path. I'm in favour of dropping them